buku

```diff
@@ -1207,6 +1216,29 @@ def prompt(results, noninteractive=False):
             break
 
 
+def prompt_delete(results, noninteractive=False):
+    """Show each matching result from a search and prompt to delete"""
+
+    count = 0
+    for row in results:
+        count += 1
+        print_record(row, count)
+
+    if noninteractive:
+        return
+
+    resp = input('Delete the search results? Enter \x1b[1my\x1b[21m to confirm: ')
+    if resp != 'y'and resp != 'Y':
+        return
+
+    # delete records in reverse order
+    pos = len(results)-1;
+    while(pos >= 0):
```

Comment:
not 100% sure either, but I doubt that the results would not be ordered by the id